src/SFA.DAS.ApplyService.Application/Services/StateService.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Reflection;
+using SFA.DAS.ApplyService.Application.Interfaces;
+
+namespace SFA.DAS.ApplyService.Application.Services
+{
+    public class StateService : IStateService
+    {
+        public Dictionary<string, object> GetState(object item)
+        {
+            var result = new Dictionary<string, object>();
+            var targetType = item.GetType();
+
+            foreach (var property in targetType.GetProperties(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic))
```

Comment:
Just to check...  _BindingFlags.NonPublic_ is fine here  (i.e. we want to know the state of anything private which would be normally hidden for reason)

Reply:
I think _NonPublic_ may not actually be required, so I will see if it can come out. I think there was a specific need for it before, when dealing with private fields within rich domain classes, but this won't be the case here.